include/matx/transforms/svd/svd_lapack.h

```diff
@@ -130,50 +130,80 @@ class matxDnSVDHostPlan_t : matxDnHostSolver_t<typename ATensor::value_type> {
     MATX_STATIC_ASSERT_STR(!is_complex_v<T3>, matxInvalidType, "S type must be real");
     MATX_STATIC_ASSERT_STR((std::is_same_v<typename inner_op_type_t<T1>::type, T3>), matxInvalidType, "A and S inner types must match");
 
-    params = GetSVDParams(u, s, vt, a, jobu, jobvt);
+    params = GetSVDParams(u, s, vt, a, jobz, algo);
     this->GetWorkspaceSize();
     this->AllocateWorkspace(params.batch_size);
   }
 
   void GetWorkspaceSize() override
   {
-    // Perform a workspace query with lwork = -1.
+    // Perform a workspace query with lwork = -1, using all mode for a
+    // larger workspace size that works for all modes
 
     lapack_int_t info;
     T1 work_query;
-    // Use all mode for a larger workspace size that works for all modes
-    gesvd_dispatch("A", "A", &params.m, &params.n, nullptr,
-                  &params.m, nullptr, nullptr, &params.m, nullptr, &params.n,
-                  &work_query, &this->lwork, nullptr, &info);
-
-    MATX_ASSERT(info == 0, matxSolverError);
-
-    // the real part of the first elem of work holds the optimal lwork.
-    // rwork has size 5*min(M,N) and is only used for complex types
-    if constexpr (is_complex_v<T1>) {
-      this->lwork = static_cast<lapack_int_t>(work_query.real());
-      this->lrwork = 5 * cuda::std::min(params.m, params.n);
+    lapack_int_t mn = cuda::std::min(params.m, params.n);
+    lapack_int_t mx = cuda::std::max(params.m, params.n);
+
+    if (params.algo == SVDHostAlgo::QR) {
+      gesvd_dispatch("A", "A", &params.m, &params.n, nullptr,
+                    &params.m, nullptr, nullptr, &params.m, nullptr, &params.n,
+                    &work_query, &this->lwork, nullptr, &info);
+
+      MATX_ASSERT(info == 0, matxSolverError);
+
+      // the real part of the first elem of work holds the optimal lwork.
+      // rwork has size 5*min(M,N) and is only used for complex types
+      if constexpr (is_complex_v<T1>) {
+        this->lwork = static_cast<lapack_int_t>(work_query.real());
+        this->lrwork = 5 * mn;
+      } else {
+        this->lwork = static_cast<lapack_int_t>(work_query);
+        this->lrwork = 0;
+      }
+    } else if (params.algo == SVDHostAlgo::DC) {
+      gesdd_dispatch("A", &params.m, &params.n, nullptr, &params.m, nullptr,
+                    nullptr, &params.m, nullptr, &params.n, &work_query,
+                    &this->lwork, nullptr, nullptr, &info);
+
+      MATX_ASSERT(info == 0, matxSolverError);
+
+      this->liwork = 8 * mn; // iwork has size 8*min(M,N) and is used for all types
```

Comment:
Are all these constants documented in the NVPL/LAPACk docs?

Reply:
These are from the LAPACK [docs](https://netlib.org/lapack/explore-html-3.6.1/d4/d84/group__complex_g_esing_gabe88d0f182417696a6e15a417911dae9.html). All of the libraries follow the same workspace requirements: